Updated react-reconciler

MarhyCZ opened the pull request:
Hi, 
I have updated react-reconciler library to support newest React.js features like React hooks for writing functional components.
Also with that I updated all the bundling dependencies like babel and webpack to their newest versions. 

Have a nice day!

MarhyCZ committed:
Updated react-reconciler

react-reconciler updated to support newest React.js features like React hooks for writing functional components.

a-ignatov-parc commented on webpack.config.js:
```diff
@@ -43,21 +42,6 @@ if (isProd) {
   plugins.push(...[
     new webpack.optimize.ModuleConcatenationPlugin(),
     new webpack.optimize.OccurrenceOrderPlugin(),
-    new UglifyJsPlugin({
```
Any reasons to remove minification for production build?

MarhyCZ replied:
Because new version of UglifyJS no longer supports ES6 and actually Webpack minifies code automatically when its in production mode. Hence the 
`mode: isProd ? 'production' : 'development',`  58th line

a-ignatov-parc replied:
Got it. Thank you 👍

a-ignatov-parc commented on webpack.config.js:
```diff
@@ -43,21 +42,6 @@ if (isProd) {
   plugins.push(...[
     new webpack.optimize.ModuleConcatenationPlugin(),
```
According to [`mode`](https://webpack.js.org/concepts/mode/) documentation `production` mode already includes `ModuleConcatenationPlugin` and `OccurrenceOrderPlugin`

MarhyCZ replied:
Oh, I forgot about that, its the same case as in removing the UglifyJS then. I will remove it then.

a-ignatov-parc commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "tvdml",
-  "version": "6.2.3",
+  "version": "6.3.0",
```
Please rollback `version` value. Releases are made with CI and they use current value

MarhyCZ replied:
Of course, no problem. will do.

a-ignatov-parc commented on package-lock.json:
```diff
@@ -0,0 +1,5962 @@
+{
```
Please delete `package-lock.json` because `yarn` is used as a default package manager

MarhyCZ replied:
sure, no problem

a-ignatov-parc reviewed (changes requested):
Thank you for your effort but I want to ask for a few changes.

Also, I need to check some issues about suspense support in `react-reconciler`

a-ignatov-parc commented:
Also, please check the tests. they are failing to run. You can run them locally with `npm test` command

MarhyCZ commented:
> Also, please check the tests. they are failing to run. You can run them locally with `npm test` command

Yeah, I see it, its just probably update from babel-register to @babel/register. I will check it all :-)

a-ignatov-parc commented:
Checked things regarding suspense. There are small changes to reconciler's config but everything should work except hiding elements which is not allowed in TVML.

Will add these changes after merging this PR 👍

MarhyCZ committed:
Fixes for new webpack, babel

MarhyCZ commented:
Ok, I updated webpack config, removed the package-lock and changed npm test script to include @babel/register not babel-register. Everything is passing now. 

Thank you for keeping this project going  💪

a-ignatov-parc reviewed (approved):
Nice! Thank you! I'll publish a new version as soon as I check update to reconciler update

a-ignatov-parc commented:
Published as `v6.3.0`
